awsume/awsumepy/lib/saml.py

```diff
@@ -41,6 +41,7 @@ def parse_assertion(assertion: str) -> list:
                 roles.append(value['#text'])
         else:
             value = attribute[attribute_value_key]
+            roles.extend(value['#text'].split(','))
             roles.append(value['#text'])
```

Comment:
not really ideal solution here but it at least fits with existing code in other places eg the something involving "if length of roles greater than 1" etc

Reply:
other regions of the code assume that "something,bar" exists in the list as opposed to only "something" and "bar" -- then also those other regions of the code require that the list also be of length 2 or greater. so it is necessary to satisfy both 

```
        if len(roles) > 1:
            if args.role_arn and args.principal_arn:
                principal_plus_role_arn = ','.join([args.principal_arn, args.role_arn])
                if self.config.get('fuzzy-match'):
                    choice = difflib.get_close_matches(principal_plus_role_arn, roles, cutoff=0)[0]
                    safe_print('Closest match: {}'.format(choice))
                else:
                    if principal_plus_role_arn not in roles:
                        raise exceptions.SAMLRoleNotFoundError(args.principal_arn, args.role_arn)
                    else:
                        choice = principal_plus_role_arn

```
    ```
  "saml2:AttributeStatement": {
        "@xmlns:saml2": "urn:oasis:names:tc:SAML:2.0:assertion",
        "saml2:Attribute": [
          {
            "@Name": "https://aws.amazon.com/SAML/Attributes/Role",
            "@NameFormat": "urn:oasis:names:tc:SAML:2.0:attrname-format:uri",
            "saml2:AttributeValue": {
              "@xmlns:xs": "http://www.w3.org/2001/XMLSchema",
              "@xmlns:xsi": "http://www.w3.org/2001/XMLSchema-instance",
              "@xsi:type": "xs:string",
              "#text": "arn:aws:iam::FAKE:saml-provider/MySamlProvider,arn:aws:iam::FAKE:role/FAKE"
            }
          },


```

Reply:
one may also ask, why do all these checks to begin with? if SAML assertion is passed, why can't we just use the role + principal from the assertion itself? what is the usecase for both supplying cli args and assertion, when the assertion has the exact same info?